src/perfrecord.cpp

```diff
@@ -144,6 +124,42 @@ void PerfRecord::record(const QStringList &perfOptions, const QString &outputPat
     m_perfRecordProcess->start(perfBinary, perfCommand);
 }
 
+void PerfRecord::record(const QStringList &perfOptions, const QString &outputPath, const QStringList &pids,
+                        const QString &workingDirectory)
+{
+    QStringList recordOptions = { QStringLiteral("--pid"), pids.join(QStringLiteral(", ")) };
```

Comment:
no need for the space after the comma, this should work too, I guess:

    pids.join(QLatin1Char(','));